operations/host_spawn.go

```diff
@@ -270,49 +291,83 @@ func hostConfigure() cli.Command {
 			if err != nil {
 				return errors.Wrapf(err, "can't find project for queue id '%s'", project)
 			}
-			cmds, err := getJasperCommands(projectRef, directory, quiet)
+
+			if distroName != "" {
+				distro, err := client.GetDistroByName(ctx, distroName)
+				if err != nil {
+					return errors.Wrap(err, "problem getting distro")
+				}
+
+				if distro.IsVirtualWorkstation {
+					if directory == cwd {
+						grip.Warning("overriding directory flag for workstation setup")
+						directory = ""
+					}
+					userHome, err := homedir.Dir()
+					if err != nil {
+						return errors.Wrap(err, "problem finding home directory")
+					}
+
+					directory = filepath.Join(userHome, evergreen.HomeVolumeDir, directory)
+				}
+			}
+
+			cmds, err := getJasperCommands(projectRef, directory, quiet, dryRun)
 			if err != nil {
 				return errors.Wrapf(err, "error getting commands")
 			}
-			for _, cmd := range cmds {
+
+			grip.Info(message.Fields{
+				"operation": "setup project",
+				"directory": directory,
+				"commands":  len(cmds),
+				"project":   projectRef.Identifier,
+				"dry-run":   dryRun,
+			})
+
+			for idx, cmd := range cmds {
 				if err := cmd.Run(ctx); err != nil {
-					gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "error running command"))
+					return errors.Wrapf(err, "problem running cmd %d of %d to provision %s", idx, len(cmds), projectRef.Identifier)
```

Comment:
Should this be idx + 1 ? Zero indexing might be confusing here, "problem running cmd 7 of 8" is misleading.

Reply:
yup